src/3d/qgs3dmapscene.cpp

```diff
@@ -486,6 +488,16 @@ void Qgs3DMapScene::updateLights()
   }
 }
 
+void Qgs3DMapScene::updateCameraLens()
+{
+  QRect viewportRect( QPoint( 0, 0 ), mEngine->size() );
+  float aspectRatio = ( float )viewportRect.width() / viewportRect.height();
+  mEngine->camera()->lens()->setFieldOfView( mMap.fieldOfView() );
+  mEngine->camera()->lens()->setPerspectiveProjection( mMap.fieldOfView(), aspectRatio, 10.f, 10000.0f );
```

Comment:
why not just `mEngine->camera()->lens()->setFieldOfView( ... )` ? We should not need to reset near/far planes and recalculate aspect ratio.  The setPerspectiveProjection() call should be just a shortcut to set multiple parameters at the same time...